dc-cudami-server/dc-cudami-server-backend-jdbi/src/main/java/de/digitalcollections/cudami/server/backend/impl/jdbi/UniqueObjectRepositoryImpl.java

```diff
@@ -1,37 +1,628 @@
+                    .mapTo(Long.class)
+                    .findOne()
+                    .get());
+    return total;
+  }
+
+  protected List<U> retrieveList(
+      String fieldsSql,
+      String fieldsSqlAdditionalJoins,
+      StringBuilder innerQuery,
+      final Map<String, Object> argumentMappings,
+      String orderBy)
+      throws RepositoryException {
+    final String sql =
+        "SELECT "
+            + fieldsSql
+            + " FROM "
+            + (innerQuery != null ? "(" + innerQuery + ")" : tableName)
+            + " AS "
+            + tableAlias
+            + (StringUtils.hasText(fieldsSqlAdditionalJoins)
+                ? " %s".formatted(fieldsSqlAdditionalJoins)
+                : "")
+            + (StringUtils.hasText(getSqlSelectReducedFieldsJoins())
+                ? " %s ".formatted(getSqlSelectReducedFieldsJoins())
+                : "")
+            + (orderBy != null && orderBy.matches("(?iu)^\\s*order by.+")
+                ? " " + orderBy
+                : (StringUtils.hasText(orderBy) ? " ORDER BY " + orderBy : ""));
+
+    // handle.execute("SET cust.code=:customerID", "bav");
+    // multitenancy, see
+    // https://varun-verma.medium.com/isolate-multi-tenant-data-in-postgresql-db-using-row-level-security-rls-bdd3089d9337
+    // https://aws.amazon.com/de/blogs/database/multi-tenant-data-isolation-with-postgresql-row-level-security/
+    // https://www.postgresql.org/docs/current/ddl-rowsecurity.html
+    // https://www.postgresql.org/docs/current/sql-createpolicy.html
+    List<U> result = execSelectForList(sql, argumentMappings);
+    return result;
+  }
+
+  public List<U> retrieveList(
+      String fieldsSql,
+      StringBuilder innerQuery,
+      final Map<String, Object> argumentMappings,
+      String orderBy)
+      throws RepositoryException {
+    // no additional joins
+    return retrieveList(fieldsSql, null, innerQuery, argumentMappings, orderBy);
+  }
+
+  protected U retrieveOne(String fieldsSql, Filtering filtering, String sqlAdditionalJoins)
+      throws RepositoryException {
+    Map<String, Object> argumentMappings = new HashMap<>(0);
+    return retrieveOne(fieldsSql, filtering, sqlAdditionalJoins, argumentMappings);
+  }
+
+  protected U retrieveOne(
+      String fieldsSql,
+      Filtering filtering,
+      String sqlAdditionalJoins,
+      Map<String, Object> argumentMappings)
+      throws RepositoryException {
+    return retrieveOne(fieldsSql, filtering, sqlAdditionalJoins, argumentMappings, null);
+  }
+
+  protected U retrieveOne(
+      String fieldsSql,
+      Filtering filtering,
+      String sqlAdditionalJoins,
+      Map<String, Object> argumentMappings,
+      String innerSelect)
+      throws RepositoryException {
+    StringBuilder sql =
+        new StringBuilder(
+            "SELECT"
+                + fieldsSql
+                + " FROM "
+                + (StringUtils.hasText(innerSelect) ? innerSelect : tableName)
```

Comment:
```suggestion
                + (StringUtils.hasText(innerSelect) ? "(%s)".formatted(innerSelect) : tableName)
```
Sollte das nicht analog zum `retrieveList` laufen?